### DISCUSSION TITLE
FPS caps itself to 60 mid gameplay

### DISCUSSION THREAD

**Discussion by gregest** [ppy/osu#24671] [Q&A]
So I've had this issue forever now, ever since lazer has been playable the past few years. I would be playing a map at hundreds of fps, unlimited being around 1k and 8x refresh rate is stable 480 and then out of nowhere it switches to 60fps which makes the game instantly unplayable because the cursor starts to lag like hell. Alt tabbing seems to fix the issue and reverts my fps to whatever I set but I can't be alt tabbing mid gameplay, restarting the level also seems to fix it but... yeah.    

The issue is not related to #24168, switching between renderers does nothing unfortunately

**Comment by peppy** [comment]
Does this happen when you move the mouse near the edge of the screen? If so, something outside of osu! may be triggering focus away from the game. I've seen this happen on my own PC before.

I presume you're on windows?

**Comment by gregest** [comment]
Nope, it happens as I'm playing the map so I'm nowhere near the edge of the screen, I am on windows yes with fullscreen

**Reply by peppy** [reply]
Hard to say then, has to be something external to the game. Try shutting down apps and maybe going through the performance guide (https://osu.ppy.sh/help/wiki/Performance_Troubleshooting)

**Reply by gregest** [reply]
I'll try doing a bunch of things then, even try switching tablet drivers because currently I use hawku drivers and who knows, might cause it. I'll go through the performace troubleshooting and change around every setting that could maybe help

**Comment by ghost** [comment]
i have the same exact problem and i was about to make a discussion about it
have you found a fix yet ?
i started playing osu lazer again yesterday on my new pc (i have this pc for 6 months) and i wasn't having this problem on my old pc but that was long time ago
this problem always happens on the first 5-10 seconds of any map with argon skin on 2x or 4x refresh only and alt tabbing instantly fix the problem but it comes back after couple of seconds
it happens sometimes on 8x refresh rate and I'm trying basically unlimited for now
i have a 75hz monitor so my fps drops to 75
i don't have any overlays or apps open like discord and i don't have geforce experience installed
i tried changing to single thread or opengl but nothing worked (this aged like milk)

after typing all of this i just tried one more thing and it fixed my problem i don't know why 
i changed renderer to opengl (legacy) and the problem is gone now and this green text wasn't there before on opengl or directx

![Screenshot 2023-12-28 045012](https://github.com/ppy/osu/assets/155034061/6a6c1026-192b-4ce2-aae5-3e6b0fead924)

Hope this helps you

**Reply by ghost** [reply]
just want to add something to this
i read a couple more discussions about this problem and i just want to provide more stuff to help this problem get fixed because i read on other discussion that osu lazer will be moving away from opengl but i am not sure if this means that opengl will be removed or not

i was on windows 10 while typing the thing at the top and i just made a clean install for windows 11 and i made sure to not install anything and let windows install the drivers. the only thing i installed is the gpu driver and i don't think its the problem because the other discussions was couple months ago and i still have the same exact problem and the only fix is still is opengl (legacy)

the problem indeed is that the game is not running on exclusive fullscreen but I'm still seeing screen tearing so this means that vsync is not capping my fps to 75 i think

whats weird is that the higher fps limit the more the problem become rare to happen. as i said on 8x (600 fps) the problem sometimes doesn't happen but on 2x or 4x it happens on the same exact time everytime and when the problem happen the game becomes unplayble like there is so much delay and i believe that is not how 75 fps feels like because i play so many games on it

![osu_2023-12-29_22-23-02](https://github.com/ppy/osu/assets/155034061/83424772-a0e4-4428-929f-2ab83f019a48)

this one as soon as the map started 
![osu_2023-12-29_22-23-33](https://github.com/ppy/osu/assets/155034061/5bbca486-6f65-4468-b20a-d4fbf7b1e5f9)

i alt tabbed to fix it
![osu_2023-12-29_22-23-42](https://github.com/ppy/osu/assets/155034061/4142e998-00b8-4222-979a-ba2cc84ac280)

the problem comes back after alt tabbing
![osu_2023-12-29_22-24-18](https://github.com/ppy/osu/assets/155034061/ade3bb2e-954c-47bf-92d7-e1319bd897a8)

[1703881353.network.log](https://github.com/ppy/osu/files/13795901/1703881353.network.log)
[1703881353.performance.log](https://github.com/ppy/osu/files/13795902/1703881353.performance.log)
[1703881353.runtime.log](https://github.com/ppy/osu/files/13795903/1703881353.runtime.log)
[1703881353.updater.log](https://github.com/ppy/osu/files/13795904/1703881353.updater.log)

**Reply by gregest** [reply]
So I know this is pretty old by now but I should answer. For me by now, after all the updates and lazer going through so many improvements since I made this report, everything seems to be fixed. I no longer experience the 60fps cap (or at least I haven't in hours of gameplay), I no longer have microstutters that I used to have and I can play at stable 1000 FPS with "basically unlimited" FPS cap and it's very smooth now. I didn't change anything at all so my assumption is that it was the game being a bit weird for me.  

For reference my pc is a gtx 1650 with i5 4690 and 16gb ram, it's very low end by todays standards
